cmp/cmpopts/sort.go

```diff
@@ -144,3 +155,126 @@ func (ms mapSorter) less(v reflect.Value, i, j int) bool {
 	}
 	return ms.fnc.Call([]reflect.Value{vx, vy})[0].Bool()
 }
+
+// GenericLess reports whether x is less than y, where both x and y must
+// be comparable types. For basic types, the values are compared by
+// value (pointer values are compared by pointer value, so ordering is
+// runtime-dependent in this case); for composite types, values are
+// compared lexicographically by component. Float and complex NaN values
+// will be compared as equal to avoid potential panics when sorting.
+//
+// GenericLess panics if provided with non-comparable values.
+func GenericLess(x, y interface{}) bool {
```

Comment:
I have concerns that will reduce the composability of options.

The fact that this is `interface{}` implies that `cmpopts.SortSlices` will be applicable on any `[]T`. Thus, the following will panic:
```go
// Hexdump transformation for better diffing.
hexdumper := cmp.Transformer(func (v []byte) []string {
    return strings.Split(hex.Dump(v), "\n")
})
cmp.Equal(x, y, cmpopts.SortSlices(nil), hexdumper)
```
This panics because it is ambiguous whether to use `SortSlices` or `hexdumper` for a value of type `[]byte`.

Reply:
Yes, that's indeed the case, but perhaps it's made a little more palatable by the fact that the panic message says immediately what the issue is and suggests how to solve it.

	cmp.Equal(x, y, cmp.FilterPath(func(p cmp.Path) bool {
		return p.Last().Type() != reflect.TypeOf([]byte(nil))
	}, cmpopts.SortSlices(nil)), hexdumper)

or perhaps:

	cmp.Equal(x, x, cmpopts.SortSlices(func(s, t string) bool {
		return cmpopts.GenericLess(s, t)
	}), hexdumper)

That is, the options are still composable with a little more effort.